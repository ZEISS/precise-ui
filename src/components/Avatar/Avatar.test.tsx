import * as React from 'react';
import * as enzyme from 'enzyme';
import { Avatar } from './';

describe('<Avatar />', () => {
  it('should render an empty <Avatar> component', () => {
    const wrapper = enzyme.mount(<Avatar />);
    expect(wrapper.find('div').length > 0).toBeTruthy();
  });

  it('should render an <Avatar> with an image', () => {
    const wrapper = enzyme.shallow(
      <Avatar image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==" />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <Avatar> with initials', () => {
    const wrapper = enzyme.shallow(<Avatar initials="FR" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <Avatar> with initials overriding the color', () => {
    const wrapper = enzyme.mount(<Avatar initials="FR" color="red" />);
    const layers = wrapper.find('div');
    expect(layers.length).toBe(3);
    expect(layers.at(2)).toHaveStyleRule('background-color', 'red');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render have a proper title in form of the description', () => {
    const wrapper = enzyme.shallow(<Avatar initials="FR" description="Sample" />);
    expect(wrapper).toMatchSnapshot();
  });
});
