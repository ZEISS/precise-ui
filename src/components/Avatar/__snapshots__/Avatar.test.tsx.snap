// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render an <Avatar> with an image 1`] = `
<styled.div
  size="medium"
>
  <styled.div
    size="medium"
  >
    <styled.img
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
    />
  </styled.div>
</styled.div>
`;

exports[`<Avatar /> should render an <Avatar> with initials 1`] = `
<styled.div
  initials="FR"
  size="medium"
>
  <styled.div
    size="medium"
  >
    <styled.div
      size="medium"
    >
      FR
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`<Avatar /> should render an <Avatar> with initials overriding the color 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  width: 40px;
  height: 40px;
}

.c1 {
  border: 0;
  -webkit-perspective: 1px;
  -moz-perspective: 1px;
  -ms-perspective: 1px;
  perspective: 1px;
  color: rgba(255,255,255,1);
  border-radius: 50%;
  overflow: hidden;
  width: 40px;
  height: 40px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  background-color: red;
  height: 40px;
  width: 40px;
  font-size: 0.875rem;
}

<Avatar
  color="red"
  initials="FR"
>
  <styled.div
    color="red"
    initials="FR"
    size="medium"
  >
    <div
      className="c0"
      color="red"
      size="medium"
    >
      <styled.div
        size="medium"
      >
        <div
          className="c1"
          size="medium"
        >
          <styled.div
            color="red"
            size="medium"
          >
            <div
              className="c2"
              color="red"
              size="medium"
            >
              FR
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Avatar>
`;

exports[`<Avatar /> should render have a proper title in form of the description 1`] = `
<styled.div
  initials="FR"
  size="medium"
  title="Sample"
>
  <styled.div
    size="medium"
  >
    <styled.div
      size="medium"
    >
      FR
    </styled.div>
  </styled.div>
</styled.div>
`;
